src/Entities/Hand.ts

```diff
@@ -1,20 +1,60 @@
-import Card from "./Card";
+import Card from './Card'
 
 class Hand {
-  private hand: Card[];
+  private hand: Card[]
 
   constructor() {
-    this.hand = [];
+    this.hand = []
   }
 
-  public getCard(cardId: string): Card {
-    // find the card that has that cardId and then remove and return it
-    throw Error("get card not implemented");
+  public addCard(card: Card): void {
+    this.hand.push(card)
   }
 
-  public addCard(card: Card): void {
-    this.hand.push(card);
+  public removeCardFromHand(cardId: string): Card {
+    if (this.hand.some((card) => card.getCardId() === cardId)) {
+      const result = this.hand.find((card) => card.getCardId() === cardId)
+      this.hand = this.hand.filter((card) => card.getCardId() !== cardId)
+      return result
+    } else {
+      throw Error(`Card id: ${cardId} not in hand`)
+    }
+  }
+
+  public getPlayableCardIds(leadCard?: Card): string[] {
```

Comment:
This function is the main one that I would like you to read over.

the reason it says `leadCard?` is because the caller is not required to pass in a `leadCard` and that will happen if the lead card hasn't been played in the trick yet, meaning that the player is about to lay the `leadCard` and they are free to play any card that they want.

Reply:
the last word!